### PR TITLE
Re-organize exports for new validators and serializers

### PR DESCRIPTION
**Pull request by adriangb**
Selected Reviewer: @hramezani

### CODE REVIEW

**Review by samuelcolvin — CHANGES_REQUESTED** [review]
I'm totally in agreement about moving methods to the new modules that make sense, but I really don't think we can remove the methods from `__init__.py`.

1. I think knowing that everything in the public API can be imported from `pydantic` is really helpful
2. One line imports for pydantic use is much cleaner and simpler
3. changing this now, especially when we're adding new recommended functions like `field_validator` will add a lot more churn to the migration

On balance I think this is one of those "correct" vs. "easy" decisions where I tend to lean towards easy and you tend to lean towards correct.

I think on this occasion we should revert that change.

The only exception to the "everything in `__init__.py`" rule right now (definitely the only important one) is `dataclass` - on this I think we should:
1. re-export our version of `dataclass`
2. rename the `dataclasses` module to `pydantic_dataclasses` to avoid confusion with std-lib dataclasses module.

**Comment by adriangb** [comment]
If we're going to export something via `pydantic/__init__.py` I don't think there's much point in also exporting it from another module. Right?

**Comment by samuelcolvin** [comment]
I think there is - some people might well want to work like

```py
from pydantic import validators

...
@validators.[tab completion]
```

More to the point, to "no export from another module" would require all other pydantic modules to be made private, which seems a little ugly.

File: pydantic/decorators.py [hunk 1]
```diff
@@ -187,81 +153,6 @@ def dec(f: Any) -> _decorators.PydanticDescriptorProxy[Any]:
...
-        # auto apply the @classmethod decorator
-        f = _decorators.ensure_classmethod_based_on_signature(f)
-
-        dec_info = _decorators.FieldValidatorDecoratorInfo(fields=fields, mode=mode, check_fields=check_fields)
-        return _decorators.PydanticDescriptorProxy(f, dec_info)
-
-    return dec
-
-
 @overload
```
Comment (samuelcolvin) [inline pydantic/decorators.py:156]: `root_validator` should surely move into `validators.py`?

More generally can we remove this module, or move it to `_internal`.

**Review by samuelcolvin — COMMENTED** [review]
otherwise LGTM.

File: pydantic/decorators.py (hunk 1 above)
Comment (adriangb) [reply pydantic/decorators.py:156]: I didn't move it because:
1. Backwards compatibility with V1.
2. We want to deprecate `@root_validator` anyway

Comment (samuelcolvin) [reply pydantic/decorators.py:156]: `root_validator` and `validator` where in `class_validators.py` in V1, the logic for supporting imports from the old path is already supported.

I think we move them.

**Comment by samuelcolvin** [comment]
I have a fix for the above issues. Please don't push until I've pushed my fix - currently can't due to problems with GH.

**Comment by samuelcolvin** [comment]
I have a fix for the above issues. Please don't push until I've pushed my fix - currently can't due to problems with GH.

**Comment by samuelcolvin** [comment]
I have a fix for the above issues. Please don't push until I've pushed my fix - currently can't due to problems with GH.

**Review by samuelcolvin — APPROVED** [review]

**Review by samuelcolvin — APPROVED** [review]

**Pull request closed without merging**